lib/lotus/model/migration.rb

```diff
@@ -0,0 +1,157 @@
+    # @see Sequel::Migration
+    Migration = ::Sequel::Migration
+
+
+    # The +Migrator+ class performs migrations based on migration files in
+    # db/migrations directory. The migration files should be named using the
+    # following pattern:
+    #
+    #   <yyyymmddhhmmss>_<title>.rb
+    #
+    # For example:
+    #
+    #   20150122124517_create_users.rb
+    #
+    # The migration file must extend +Lotus::Model::Migration+ and
+    # implement +up+ and +down+ methods.
+    # @see Lotus::Model::Migration
+    #
+    # To apply a migrator, the +apply+ must be invoked with the
+    # SQL adapter config instance, and optional target version.
+    # If no current version is supplied, it is read from the database.
+    # The migrator automatically create schema_migration table to
+    # keep track of current migration version. If no migration is
+    # stored in the database, the version is considered to be 0.
+    # if no target version is specified, the database is migrated
+    # to the latest version available in db/migration directory.
+    #
+    # @example Migrate all migrations in db/migrations to latest
+    #   require 'lotus/model/migration'
+    #   adapter_cfg = Lotus::Model::Config::Adapter.new(type: :sql, uri: DATABASE_URI)
+    #   Lotus::Model::Migrator.new(adapter_cfg).run
+    #
+    # @example To use shortcut +apply+ method to invoke +run+ instance method
+    #   require 'lotus/model/migration'
+    #   adapter_cfg = Lotus::Model::Config::Adapter.new(type: :sql, uri: DATABASE_URI)
+    #   Lotus::Model::Migrator.apply(adapter_cfg)
+    #
+    # @since x.x.x
+    #
+    # @see Sequel::Migration
+    # @see Sequel::TimestampMigrator
+    class Migrator
+      # It's raised when migration does not support the adapter
+      class UnsupportedAdapter < ::StandardError
+        def initialize
+          super("Migrator only supports SQL adapter")
+        end
+      end
+
+      extend Forwardable
+      delegate [:run, :is_current?] => :@migrator
+
+      MIGRATION_DIRECTORY = "#{Dir.pwd}/db/migrations".freeze
+
+      # Initialize a migrator instance
+      #
+      # Default logging to +Lotus::Logger+
+      #
+      # @param adapter_config [Lotus::Model::Config::Adapter] a SQL adapter config
+      # @param directory [String] the location of migration files (default: db/migrations)
+      # @param target [String] the target migration file
+      # @param current [String] the current migration version
+      #
+      # @return [Lotus::Model::Migrator] a new migrator instance
+      #
+      # @since x.x.x
+      #
+      # @see Sequel::TimestampMigrator
+      def initialize(adapter_config, directory: MIGRATION_DIRECTORY, target: nil, current: nil, logger: ::Logger.new(STDOUT))
+        raise Lotus::Model::Migrator::UnsupportedAdapter if adapter_config.type != :sql
+        @db = ::Sequel.connect(adapter_config.uri)
+        @migrator = ::Sequel::TimestampMigrator.new(db, directory, {target: target, current: current})
```

Comment:
I am going to stick without spaces because it is now a convention that we follow ;)